content/protocols/mqtt.textile

```diff
@@ -0,0 +1,236 @@
+});
+```
+
+Or, if you wish to decode messages received through MQTT, you can use the following in NodeJS:
+
+```[nodejs]
+const encoding = require('text-encoding');
+var decoder = new encoding.TextDecoder();
+var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+
+client.on('connect', function () {
+  client.subscribe('my_channel');
+});
+
+client.on('message', function (topic, message) {
+  console.log(decoder.decode(message));
+});
+```
+
+In the above example, @command@ will now contain the message in its original string form.
+
+h2(#ssl-tls). SSL/TLS
+
+Ably supports both SSL and non-SSL connections (the latter uses a different port, see above), but strongly recommend using SSL wherever possible. If you are not using SSL, note that the same restrictions apply as if you were using an Ably client without SSL. That is, "connection attempts using Basic Authentication (i.e. an API key) are disallowed":/auth/basic, and any "namespaces which you've enabled 'require TLS' on":/channels will be inaccessible. "Contact us":https://ably.com/contact if this is a problem for you.
+
+h2(#channel-options). Channel options
+
+In an MQTT connection, it is also possible to specify "channel options":/channels/options as a query string in the connection URL, instead of as a qualifier on an individual channel name. In this case, the given channel options apply to all channel attachments associated with that connection.
+
+Currently both "rewind":/channels/options/rewind and "delta":/channels/options/delta channel options are supported.
+
+h3(#delta-mqtt). Delta with MQTT
+
+If subscribing to a channel in delta mode using MQTT or one of the protocol adaptors such as "SSE":/protocols/sse, then you will need to decode any received delta messages yourself. There are decoder libraries available to do this for several platforms; see the "download":LINK-ME section for details.
+
+When subscribing without an Ably library, the channel @delta@ parameter must be specified using a "qualified channel name":/channels/options. 
+
+Some transports provide raw message payloads - that is, the content of the @data@ attribute of a @Message@ - without the accompanying metadata. That means that the recipient of the message does not have access to the @extras@ or @encoding@ attributes of the message that would ordinarily be used to decode delta message payloads.
+
+Examples of such transports are MQTT, and SSE in non-enveloped mode. In order to assist applications that use these transports, the @vcdiff@ decoder libraries can check for the @vcdiff@ header at the start of the message payload as an inexact method of determining whether or not the message is a regular message or a delta. Note that, in order to rely on that check, you need to know that that header will not be present in any valid (uncompressed) message in your app. No valid JSON value, for example, will match the @vcdiff@ header check, so it is safe to perform this sniffing on JSON message payloads.
```

Comment:
Suggest we make this one specific to MQTT again too, please. The info will need to be repeated in SSE where applicable.